### PR TITLE
Use cro to deploy blobstorage for 3scale

### PR DESCRIPTION
**Pull request by dimitraz**
## Description
Use cro to generate blobstorage to be used by 3scale.   

## Verification 
- Create an operator source pointing to the integreatly registry 
```yml
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: integreatly
  type: appregistry
```
- Create a configmap in the integreatly namespace for the cloud resource operator which specifies in-cluster only deployments: 
```yml
kind: ConfigMap
apiVersion: v1
metadata:
  name: cloud-resource-config
  namespace: intly
data:
  managed: |
    {"blobstorage":"openshift", "smtpcredentials":"openshift", "redis":"openshift", "postgres":"openshift"}
  workshop: |
    {"blobstorage":"openshift", "smtpcredentials":"openshift", "redis":"openshift", "postgres":"openshift"}
```
- Create a configmap in the integreatly namespace for openshift provider specific configuration:
```yml
kind: ConfigMap
apiVersion: v1
metadata:
  name: cloud-resources-openshift-strategies
data:
  blobstorage: |
    {"production": { "strategy": {} }}
  smtpcredentials: |
    {"production": { "strategy": {} }}
```
- Run the integreatly operator from this branch and ensure the `cloud-resources` stage completes
- Ensure that 3scale is installed correctly
- Ensure that a `BlobStorage` custom resource and the respective secret with connection details to this resource are created in the integreatly namespace
- Ensure that an `s3-credentials` secret is created in the 3scale namespace with the same credentials that are in the blob storage secret

### CODE REVIEW

File: pkg/controller/installation/products/threescale/reconciler.go [hunk 1]
```diff
@@ -258,45 +264,12 @@ func (r *Reconciler) reconcileSMTPCredentials(ctx context.Context, inst *v1alpha
...
 
 func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient pkgclient.Client) (v1alpha1.StatusPhase, error) {
-	bucket := &v1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: s3BucketSecretName,
-		},
-	}
-
-	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: bucket.Name, Namespace: r.ConfigManager.GetOperatorNamespace()}, bucket)
+	fss, err := r.getBlobStorageFileStorageSpec(ctx, serverClient)
```
Comment (aidenkeating) [inline pkg/controller/installation/products/threescale/reconciler.go]: do any of the README docs need to change to account for these changes. for example, we currently tell a user to pre-seed a secret, is that required anymore, and if they do pre-seed it, is it used?

Comment (dimitraz) [reply pkg/controller/installation/products/threescale/reconciler.go]: Ok will update :+1:

**Comment by ciaranRoche** [comment]
@dimitraz does this need another verification?

**Comment by austincunningham** [comment]
/lgtm
will verify now

**Pull request merged**